[pages] Add Bulk Redirect How To for Default pages.dev (#9250)
cloudflare/cloudflare-docs: content/pages/how-to/redirect-to-custom-domain.md

---
pcx_content_type: how-to
title: Redirecting *.pages.dev to a Custom Domain
---

# Redirecting *.pages.dev to a Custom Domain

In this guide, you will learn how to redirect your `*.pages.dev` subdomain to your Custom Domain (`example.com`). This is a common question and can be achieved by using [Bulk Redirects](/rules/url-forwarding/bulk-redirects/).

To do this:

> 1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/pages/view/:pages-project/domains), and select your account and Pages Project.
> 2. Ensure that your custom domain is listed in the **Custom domains** section. If it is not, add it by clicking **Set up a custom domain**.

Reviewer: ```suggestion
1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/pages/view/:pages-project/domains), and select your account.
2. Select **Workers & Pages** and select your Pages application.
3. Go to **Custom domains** and make sure that your custom domain is listed. If it is not, add it by clicking **Set up a custom domain**.
```

> 
> ![Ensure custom domain has been added to Pages](/pages/how-to/media/custom-domain-dash.png)

Reviewer: ```suggestion
```

Text description is sufficient. We can delete this file from the branch.

> 
> 3. Next, go to **Account Home** > **Bulk Redirects** > **Create a new Bulk Redirects list** > **Create new list**.

Reviewer: ```suggestion
4. Go to **Account Home** > **Bulk Redirects**.
5. Select **Create a new Bulk Redirects list** > **Create new list**.
```

![Create a new Bulk redirects list](/pages/how-to/media/create_a_new_bulk_redirect_list.png)

> 4. In the content type, select **Redirect**. 
> 5. Add your `*.pages.dev` Project Subdomain as your source URL.
> 6. Add your target custom domain URL. Note that you must include `https://` before the apex domain. 
> 7. Select **Edit parameters** > select **Preserve query string**, **Subpath matching** and **Preserve path suffix**.
> 8. *Optional*: Tick the **Include subdomains** box. This will automatically redirect all of the preview URLs to your custom domain.

Reviewer: ```suggestion
6. In the content type, select **Redirect**. 
7. Add your application's `*.pages.dev` subdomain as your source URL.
8. Add your target custom domain URL. Note that you must include `https://` before the apex domain. 
9. Select **Edit parameters** and select **Preserve query string**, **Subpath matching** and **Preserve path suffix**.

*Optional*: Tick the **Include subdomains** box. This will automatically redirect all of the preview URLs to your custom domain.
```

> 
> ![Follow the steps above to correctly configure your list](/pages/how-to/media/pages-dev-redirect-parameters.png)
> 

Reviewer: ```suggestion
```
Same comment here, the text instructions are sufficient.

> 9. Select **Add to list**.
> 10. Go to **Bulk Redirects** > **Create Bulk Redirects** > select your list > **Save and Deploy**.

Reviewer: ```suggestion
10. Select **Add to list**.
11. Go to **Bulk Redirects** > **Create Bulk Redirects** > select your list > **Save and Deploy**.
```

---
pcx_content_type: how-to
title: Redirecting *.pages.dev to a Custom Domain
---

# Redirecting *.pages.dev to a Custom Domain

In this guide, you will learn how to redirect your `*.pages.dev` subdomain to your Custom Domain (`example.com`). This is a common question and can be achieved by using [Bulk Redirects](/rules/url-forwarding/bulk-redirects/).

To do this:

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/?to=/:account/pages/view/:pages-project/domains), and select your account and Pages Project.
2. Ensure that your custom domain is listed in the **Custom domains** section. If it is not, add it by clicking **Set up a custom domain**.

![Ensure custom domain has been added to Pages](/pages/how-to/media/custom-domain-dash.png)

> 3. Verify that Access has not been enabled on your Pages Project. While you can apply Access policies to your Custom Domain, if you enable them on your `*.pages.dev` domain, then your custom domain will not function. You can view Access Policies by going to the [Zero Trust Dash](https://one.dash.cloudflare.com), selecting your account, and going to **Access** > **Applications**.

Reviewer 2: I'm not sure we really need this step, if you have done this, then you've done it for both pages.dev + domain so it's ok.

By default, Access only covers previews.
